[Feature] Update for TYPO3 v13

DavidBruchmann opened the pull request:
This PR replaces #53 and has better sorted and explained commits / commit messages and fix #54.

All known issues have been fixed in this branch.

The PR is based on the main branch and shall be used to merge into the new branch (to be created) `TYPO3_13` from there.

I don't tag this, as I've no access to publish automatically to TER.
The version is 4.0.0 and should be tagged as well as getting a release.

My version constraints in composer.json look a bit clumsy, let me know what you think, example:
```
"typo3/cms-backend": "^13.3.1 || ^13.4 || dev-main",
```

PKuhlmay commented on composer.json:
```diff
@@ -23,41 +23,41 @@
 		"php": "^8.1",
-		"typo3/cms-backend": "^12.4",
-		"typo3/cms-core": "^12.4"
+		"typo3/cms-backend": "^13.3.1 || ^13.4 || dev-main",
```
IMO the declaration of `^13.4` is not necessary here because the `^13.3.1` should include the version 13.4. After the release of ^13.4 next week, this could be adjusted then.

DavidBruchmann replied:
yeah, I thought it looks terrible, LOL.
Waiting for another opinion.

PKuhlmay commented on Resources/Public/JavaScript/paste-reference.js:
```diff
@@ -75,17 +81,17 @@ const onReady = new OnReady;
-  const url = $element.data('url') || null;
-  const elementTitle = top.itemOnClipboardTitle != undefined ? top.itemOnClipboardTitle : "["+TYPO3.lang['tx_paste_reference_js.modal.labels.no_title']+"]";
+Paste.activatePasteModal = (function (element) {
+  // const $element = $(element);
```
Is this still needed? As it is commented out.

PKuhlmay commented:
The button "Copy content from another page to this place" is not working for me. The modal is opening, but then I cannot add something as a reference.

DavidBruchmann commented:
> The button "Copy content from another page to this place" is not working for me. The modal is opening, but then I cannot add something as a reference.

Mhm, checking again, actually I tested it ...

DavidBruchmann commented:
@PKuhlmay I updated my TYPO3 installation and tested again, I can't reproduce you issue, so for me it's working fine.
I also deleted my code to fetch it new from the repository for testing.

PKuhlmay commented:
> @PKuhlmay I updated my TYPO3 installation and tested again, I can't reproduce your issue, so for me it's working fine. I also deleted my code to fetch it new from the repository for testing.

OKay, maybe it is a problem with my installation. I will test this later in a clean one.

DavidBruchmann commented:
That would be great!
Thanks a lot!!!

sbuerk commented:
Set to draft to make clear that this needs to wait until main / v12 has been finished first.

Kephson reviewed (approved):
Approved to merge the changes into a TYPO3_13 branch
